pkg/reconciler/v1alpha1/clusteringress/clusteringress.go

```diff
@@ -370,22 +369,24 @@ func (c *Reconciler) ensureFinalizer(ci *v1alpha1.ClusterIngress) error {
 func (c *Reconciler) reconcileDeletion(ctx context.Context, ci *v1alpha1.ClusterIngress) error {
 	logger := logging.FromContext(ctx)
 
+	finalizers := sets.NewString(ci.Finalizers...)
 	// If our Finalizer is first, delete the `Servers` from Gateway for this ClusterIngress,
 	// and remove the finalizer.
-	if len(ci.Finalizers) == 0 || ci.Finalizers[0] != clusterIngressFinalizer {
+	if !finalizers.Has(clusterIngressFinalizer) {
```

Comment:
This is now incorrect.  We should ONLY process our finalizer when we are "next" (first).

Reply:
So if the order matters, we can't do `Sets(x).List()` above either.

Reply:
I see. Thanks!
Now I use `append` instead of doing `Sets(x).List()` in order to keep the order of finalizers.

We also need to do the similar thing for the [finalizers of Route](https://github.com/knative/serving/blob/master/pkg/reconciler/v1alpha1/route/route.go#L395).